components/bl-mention-component/src/index.js

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState } from 'react';
+
+import primereact from './lib/core';
+
+const { Mention } = primereact.mention;
+const { cn } = BackendlessUI.CSSUtils;
+
+const DEFAULT_FIELD_BLACK_LIST = ['created', '___class', 'ownerId', 'updated', 'objectId'];
+
+export default function MentionComponent({ component, eventHandlers }) {
+  const {
+    trigger, suggestions, field, scrollHeight, autoHighlight, placeholder, delay, autoresize, rows, cols,
+    hideField, classList, style, display,
+  } = component;
+  const { onChange, onFocus, onBlur, onShow, onHide } = eventHandlers;
+
+  const [processedSuggestions, setProcessedSuggestions] = useState([]);
+  const triggers = useTriggers(trigger);
+  const [fieldBlackList, setFieldBlackList] = useState(DEFAULT_FIELD_BLACK_LIST);
+
+  useEffect(() => {
+    if (hideField?.length) {
+      const hideValues = hideField.split(',').map(el => el.trim());
+      setFieldBlackList(prev => [...prev, ...hideValues]);
+    }
+  }, [hideField]);
+
+  const searchHandler = event => {
+    const eventTrigger = event.trigger;
+
+    for (const trigger of triggers) {
+      if (trigger !== eventTrigger) {
+        continue;
+      }
+
+      const query = event.query;
+      const orderedSuggestions = suggestions.filter(suggestion => suggestion.trigger === eventTrigger)[0].suggestions;
+      let finalSuggestions;
+
+      if (!query.trim().length) {
+        finalSuggestions = orderedSuggestions;
+      } else {
+        finalSuggestions = orderedSuggestions
+          .filter(suggestion => suggestion.field.toLowerCase().startsWith(query.toLowerCase()));
+      }
+
+      setProcessedSuggestions(finalSuggestions);
+    }
+  };
+
+  const itemTemplate = (suggestion, options) => {
+    const trigger = options.trigger;
+    const fields = [];
+
+    for (const field in suggestion) {
+      if (!fieldBlackList.includes(field)) {
+        fields.push(field);
+      }
+    }
+
+    if (fields.indexOf('img') > 0) {
+      const index = fields.indexOf('img');
+      [fields[0], fields[index]] = [fields[index], fields[0]];
+    }
+
+    const availableTrigger = triggers.some(availableTrigger => availableTrigger === trigger);
```

Comment:
availableTrigger -  means we have an entity that is available. But method some returns boolean value.
So

If it realy should be 'availableTrigger' - use find method(for example)
But if it should be boolean - rename it to hasAvailableTrigger / hasTrigger / isTriggerAvailable